test/extended/networking/util.go

```diff
@@ -112,75 +97,42 @@ func providerIs(providers ...string) bool {
 	return false
 }
 
-// Filters nodes in NodeList in place, removing nodes that do not
-// satisfy the given condition
-// TODO: consider merging with pkg/client/cache.NodeLister
-func filterNodes(nodeList *api.NodeList, fn func(node api.Node) bool) {
-	var l []api.Node
-
-	for _, node := range nodeList.Items {
-		if fn(node) {
-			l = append(l, node)
-		}
-	}
-	nodeList.Items = l
-}
-
-func getMultipleNodes(f *e2e.Framework) (nodes *api.NodeList) {
-	nodes, err := f.Client.Nodes().List(api.ListOptions{})
-	if err != nil {
-		e2e.Failf("Failed to list nodes: %v", err)
-	}
-	// previous tests may have cause failures of some nodes. Let's skip
-	// 'Not Ready' nodes, just in case (there is no need to fail the test).
-	filterNodes(nodes, func(node api.Node) bool {
-		return isNodeReadySetAsExpected(&node, true)
-	})
-
-	if len(nodes.Items) == 0 {
-		e2e.Failf("No Ready nodes found.")
-	}
-	if len(nodes.Items) == 1 {
-		// in general, the test requires two nodes. But for local development, often a one node cluster
-		// is created, for simplicity and speed. (see issue #10012). We permit one-node test
-		// only in some cases
-		if !providerIs("local") {
-			e2e.Failf(fmt.Sprintf("The test requires two Ready nodes on %s, but found just one.", exutil.TestContext.Provider))
-		}
-		e2e.Logf("Only one ready node is detected. The test has limited scope in such setting. " +
-			"Rerun it with at least two nodes to get complete coverage.")
-	}
-	return
-}
+func launchWebserverService(f *e2e.Framework, serviceName string, nodeName string) (serviceIP string) {
+	e2e.LaunchWebserverPod(f, serviceName, nodeName)
+	// FIXME: make e2e.LaunchWebserverPod() set the label when creating the pod
+	podClient := f.Client.Pods(f.Namespace.Name)
+	pod, err := podClient.Get(serviceName)
+	expectNoError(err)
+	pod.ObjectMeta.Labels = make(map[string]string)
+	pod.ObjectMeta.Labels["name"] = "web"
```

Comment:
Consider defining the service first and copying the label to the pod rather than defining independently.